app/components/FolderContentScreen.tsx

```diff
@@ -37,23 +54,165 @@ const FolderContentScreen = (props: NativeStackScreenProps<RootStackParamList, "
+                        animationConfig={{
+                            duration: 300
+                        }}
+                        from={(
+                            <TouchableOpacity style={styles.uploadButton} onPress={() => setSettingsPopupVisible(true)}>
+                                <Feather name="plus" size={40} color="#fff" />
+                            </TouchableOpacity>
+                    )}>
+                        <View style={styles.settingModal}>
+                            <TouchableOpacity style={styles.settingButton} onPress={() => uploadFileHandler()}>
+                                <AntDesign name="addfile" size={25} color={Colors.primary} />
+                                <Text style={styles.settingText}>NEUE DATEI</Text>
+                            </TouchableOpacity>
+                            <TouchableOpacity style={styles.settingButton} onPress={() => {
+                                setTimeout(() => {
+                                    // For some reason iOS needs a second before it registers the modal.
+                                    // If we don't wait the modal is invisible.
+                                    setModalVisible(true)
+                                }, 1000)
+                                setSettingsPopupVisible(false);
+                            }}>
+                                <AntDesign name="addfolder" size={25} color={Colors.primary} />
+                                <Text style={styles.settingText}>NEUER ORDNER</Text>
+                            </TouchableOpacity>
+                        </View>
+                    </Popover>
+                :
+                    null
+            }
+
+            <Modal
+                animationType="slide"
+                transparent={true}
+                visible={modalVisible}
+                onRequestClose={() => setModalVisible(false)}
             >
-            <FileList folders={allFolders} files={allFiles} refreshFunction={fetchFolderContent}/>
-        </ScrollView>     
+                <View style={styles.modalOverlay}>
+                    <View style={styles.modalContainer}>
+
+                        <TouchableOpacity
+                            style={styles.closeModal}
+                            onPress={() => {
+                                setModalVisible(false);
+                            }}
+                        >
+                            <Feather name="x" size={20} color="black" />
+                        </TouchableOpacity>
+
+                        <Text style={styles.modalTitle}>Ordner Name</Text>
+                        <TextInput
+                            style={styles.textInput}
+                            placeholder="Wähle einen Ordner Namen..."
+                            value={newFolderName}
+                            onChangeText={(text: string) => {
+                                if (/[^a-zA-Z0-9-_ ]/.test(text)) {
```

Comment:
Hmm, it's not so easy... I would also like to have Ö ö Ä ä Ü ü and ß in my foldername.

Reply:
ahh I simply forgot, one second

Reply:
now all kind of letters are allowed, I also set a max folder name length (128).